pkgs/applications/science/electronics/xyce/default.nix

```
@@ -0,0 +1,159 @@
+{ stdenv
+, fetchgit
+, fetchurl
+, lib
+, autoconf
+, automake
+, bison
+, blas
+, flex
+, fftw
+, gfortran
+, lapack
+, libtool_2
+, mpi
+, suitesparse
+, trilinos # if compiling with MPI so trilinos must be! Pass the correct one.
+, withMPI ? false
+  # for doc
+, texlive
+, pandoc
+, enableDocs ? true
+  # for tests
+, bash
+, bc
+, openssh # required by MPI
+, perl
+, perlPackages
+, python38
+, enableTests ? true
+}:
+
+stdenv.mkDerivation rec {
+
+  pname = "xyce";
+  version = "7.4.0";
+  twoLevelVersion = with builtins;
+    concatStringsSep  "." (lib.lists.take 2 (splitVersion version));
+
+  srcs = [
+    # useing fetchurl or fetchFromGitHub doesn't include the manuals
+    # due to .gitattributes files
+    (fetchgit {
+      url = "https://github.com/Xyce/Xyce.git";
+      rev = "82f96bbe05bac5921cd6fa1e8bb6a2983a797bf8";
+      sha256 = "sha256-sOHjQEo4FqlDseTtxFVdLa0SI/VAf2OkwQV7QSL7SNM=";
+    })
+    (fetchurl {
+      url = "https://github.com/Xyce/Xyce_Regression/archive/refs/tags/Release-${version}.tar.gz";
+      sha256 = "00wy2kyvbsyn6gfy42lrkzvgfq9k89hli3ryckywj6p1kcwnvdn9";
+    })
+  ];
+
+  sourceRoot = with builtins; "./Xyce-${substring 0 7 (head srcs).rev}";
+
+  preConfigure = "./bootstrap";
+
+  configureFlags = [ "CXXFLAGS=-O3"
+                     "--enable-xyce-shareable"
+                     "--enable-shared"
+                     "--enable-stokhos"
+                     "--enable-amesos2"
+                   ] ++ lib.optionals withMPI [
+                     "--enable-mpi"
+                     "CXX=mpicxx"
+                     "CC=mpicc"
+                     "F77=mpif77"
+                   ];
+
+  nativeBuildInputs = [ autoconf automake gfortran libtool_2
+                      ] ++ lib.optionals enableDocs [
+                        (texlive.combine {
+                          inherit (texlive)
+                            scheme-medium
+                            koma-script
+                            optional
+                            framed
+                            enumitem
+                            multirow
+                            preprint; })
+                      ];
```

```suggestion
  nativeBuildInputs = [
    autoconf automake gfortran libtool_2
  ] ++ lib.optionals enableDocs [
    (texlive.combine {
      inherit (texlive)
        scheme-medium
        koma-script
        optional
        framed
        enumitem
        multirow
        preprint;
      })
  ];
```
Is there no already build texlive package that is close enough to what we need? If so please use that.

---

I've tried several packages, but have not found one working.